hazelcast/src/main/java/com/hazelcast/spi/impl/NodeEngine.java

```diff
@@ -142,12 +150,18 @@ public interface NodeEngine {
     Config getConfig();
 
     /**
-     * Returns the Config ClassLoader. This class loader will be used for instantiation of all classes defined by the
-     * configuration (e.g. listeners, policies, stores, partitioning strategies, split brain protection functions, ...).
+     * Returns the Config ClassLoader. This class loader will be used for the instantiation of all classes defined
+     * by the configuration (e.g. listeners, policies, stores, partitioning strategies, split brain protection
+     * functions, etc.).
+     * <p>
+     * When the {@link NamespaceService} is enabled, this will return an instance of the
+     * {@link com.hazelcast.internal.namespace.impl.NamespaceAwareClassLoader}, which delegates to child loaders
+     * depending on the Namespace context of the class loading.
```

Comment:
No, that's possible - the `NamespaceAwareClassLoader` builds on top of the config classloader, so it will fallback to that if  it's unable to find a Namespace for the data structure.